api/src/test/resources/io/strimzi/api/kafka/model/KafkaConnector.out.yaml

```diff
@@ -3,11 +3,11 @@ apiVersion: "kafka.strimzi.io/v1alpha1"
 kind: "KafkaConnector"
 metadata:
   name: "test-source"
+  labels:
+    connect-cluster: my-connect-cluster
 spec:
   class: "org.apache.kafka.connect.file.FileStreamSourceConnector"
   tasksMax: 2
-  topics:
-  - "test.topic"
   config:
-  - name: "file"
-    value: "/home/source/test.source.txt"
\ No newline at end of file
+    file: "/home/source/test.source.txt"
+    topic: "test.topic"
```

Comment:
didn't we decide to stay with name-value reading the related #2037 ? @scholzj @tombentley

Reply:
I can see the advantage in both ... so I can probably live with this easily.

Reply:
@scholzj please define "this".

Reply:
With _this_ I meant the map which is used in this PR. I can live with both TBH ;-), I see some pros and cons in both and do not have strong opinion.

Reply:
Yeah, I'm similarly conflicted. I can see that the name-value approach inhibits copy+paste. But API flexibility (when you need it) is priceless. The question is: do we _really_ need it? We have a mechanism for mounting secrets etc into the container, what use case isn't already covered by that?

Reply:
I think that mounting the secrets into connect and using the providers is (in my naivity) more secure because it does better layering. The person managing the connectors doesn't get a way to access the secrets without any RBAC protection (only the person managing Kafkaconnect can do that :-/). Obviously if you leave this aside, it is more user friendly to do everything in single place.

The question is also how would you implement it techncially: proper way would be to mount them into the connect and use the config provider. But that will be hard to implement with separate ConnectorOperator and ConnectOperator. It would also mean Connect restarts just because someone change a connector config which means disruption for other connectors running in the same Connect.

Getting the values directly in CO and passing them in plaintext config on the HTTP interface would mean that anyone who gets access to the HTTP REST interface and lists the connectors sees the secrets.

So I don't know ... I'm not convinced it is the right way. But if we do the map now it will be hard to change our mind later. If we go with `key` and `value` fields, we can decide that later but at the price of worse UX. 🤯

Did Gunnar as a someone more familiar with practical usage of connectors ever commented on this?

Reply:
I think Gunnar favoured `foo: bar` over `name: foo, value: bar`, but I'm not sure whether whether we was aware of the the argument for flexibility in the API. But let's ask him: @gunnarmorling.

`name: foo, value: bar` is also inconsistent with `Kafka.spec.kafka.config` etc. I think we should aim for "eventual consistency", so if we thought we would benefit from `name: foo, value: bar` elsewhere that could also be a reason to stick with it here.